### PR TITLE
Enhanced neighbor point filter

### LINKED ISSUE #24: Implement extrication trajectory filter

### PR DESCRIPTION
**Pull request by Nandite**
@VictorLamoine @KevinBollore , tested on planes, cylinders and DAC, working well.

Fixes #24

### CODE REVIEW

**Comment by VictorLamoine** [comment]
Libasse, you have merge conflicts with this branch so the commit can't be merged.

Here is what you need to do:
- Save your work (branch, zip or whatever, just make sure you won't loose anything)
- Create a new branch at indido-devel-experimental state
- Cherry pick this commit: d5580dcc4954c24ad18938092050bdf0405ed945
`git cherry-pick d5580dcc4954c24ad18938092050bdf0405ed945`
- Cherry pick your new commit:
`git cherry-pick 90583f5fe2985a7b5eeaffd7fde3b6619ecdbfed`

Then rebase against indigo-devel-experimental and sqash the two commits into one.

Tell me if you fail to do it!

**Comment by Nandite** [comment]
It's ok ! @KevinBollore gave me a hand :wave:

**Comment by VictorLamoine** [comment]
Yup its perfect now!

@KevinBollore can you test and give me feedback? is it safe to merge?

File: bezier_library/src/bezier_grinding_surfacing.cpp [hunk 1]
```diff
@@ -430,7 +430,11 @@ bool BezierGrindingSurfacing::generateRobotPosesAlongStripper(const vtkSmartPoin
     }
   }
 
-  filterNeighborPosesTooClose(point_normal_table, 1e-4);
+  if(!filterNeighborPosesTooClose(point_normal_table, 1e-2))
+  {
+    ROS_ERROR_STREAM("BezierGrindingSurfacing::generateRobotPosesAlongStripper: Filter failure");
```
Comment (VictorLamoine) [inline bezier_library/src/bezier_grinding_surfacing.cpp]: `Neighbor filtering failure`?

This is not very explicit

Comment (Nandite) [inline bezier_library/src/bezier_grinding_surfacing.cpp]: Corrected ! :rabbit:

Comment (VictorLamoine) [inline bezier_library/src/bezier_grinding_surfacing.cpp]: `grinding trajectory filtering failure`?

The idea is to be able to differentiate if the grinding or extrication path filtering failed.

**Comment by Nandite** [comment]
`cannot filter grinding trajectory` will be ok ?

**Comment by KevinBollore** [comment]
It looks great and works... But, only on scan_defect.ply, there is a strange behavior on the last point of some lines...

image

**Comment by VictorLamoine** [comment]
Ok so the before last pose isn't filtered, why?
The normal is not correct but I guess that is due to issue #8

**Comment by VictorLamoine** [comment]
@Nandite yes

**Comment by KevinBollore** [comment]
Yes, and issue #8 is fixed in PR #29.
I'll mix PR #29 and this one locally and check the behavior

**Comment by KevinBollore** [comment]
Fix #27

**Pull request merged**